rust/otap-dataflow/crates/otap/src/syslog_cef_receiver.rs

```diff
@@ -330,9 +422,23 @@ impl local::Receiver<OtapPdata> for SyslogCefReceiver {
                                         // Reset the timer since we already built an arrow record batch due to size constraint
                                         interval.reset();
 
-                                        effect_handler.send_message(OtapPdata::new_todo_context(arrow_records.into())).await?;
+                                        let res = effect_handler.send_message(OtapPdata::new_todo_context(arrow_records.into())).await;
+                                        {
+                                            let mut m = self.metrics.borrow_mut();
+                                            match &res {
+                                                Ok(_) => m.received_logs_forwarded.add(items),
+                                                Err(_) => m.received_logs_forward_failed.add(items),
+                                            }
+                                        }
+                                        // UDP path: do not propagate downstream send errors; keep running
+                                        // so that telemetry can still be collected (tests expect refused
+                                        // to be counted and reported). We already incremented
+                                        // received_logs_refused above.
```

Comment:
Ah so if you're already applying the fix for UDP in this PR, I'd prefer we apply it for TCP as well in this PR. For both TCP and UDP we want the task to keep running.

```suggestion
                                        // Do not propagate downstream send errors; keep running
                                        // so that telemetry can still be collected (tests expect refused
                                        // to be counted and reported). We already incremented
                                        // `received_logs_refused` above.
```